docs/specs/mlnet-cli/MLNET-CLI-Specs.md

```diff
@@ -0,0 +1,969 @@
+# Specs for the ML.NET CLI with AutoML
+
+## Context
+
+**Commitments: This specs document is 100% aspirational and will change while it's being discussed and implementation is evolving based on feedback. There are no commitments derived from this document except for the first upcoming minor version at any given time (v0.1, initially).**
```

Comment:
>v0.1, [](start = 264, length = 5)

If CLI is bound to specific version of ML.NET, consider synchronizing the version in some way to know that this version of CLI works for a given version of  ML.NET. For examle, consider calling CLI for ML.NET v0.10 to be CLI v0.10

Reply:
Coupling like that can't happen unless we want to always tie the versions together. For example, when we ship ML.NET 1.0, will the CLI be `1.0`? I assume not... But I agree it does become a version zoo (we on the .NET team know...), but it is necessary until you can sync up the schedules.

See https://github.com/dotnet/designs/blob/master/accepted/sdk-version-scheme.md for how .NET Core tackles this problem.